api/app/routers/hfi_calc.py

```diff
@@ -79,100 +93,121 @@ def extract_selected_stations(request: HFIResultRequest) -> List[int]:
+                # Fetching dailies is an expensive operation. When a user is clicking an unclicking stations
+                # in the front end, we'd prefer to not change the the call that's going to wfwx so that we can
+                # use cached values. So we don't actually filter out the "selected" stations, but rather go
+                # get all the stations for this fire centre.
+                fire_centre_stations = get_fire_centre_stations(
+                    orm_session, request.selected_fire_center_id)
+                fire_centre_station_code_ids = set()
+                area_station_map = {}
+                station_fuel_type_map = {}
+                for station, fuel_type in fire_centre_stations:
+                    fire_centre_station_code_ids.add(station.station_code)
+                    if not station.planning_area_id in area_station_map:
+                        area_station_map[station.planning_area_id] = []
+                    area_station_map[station.planning_area_id].append(station)
+                    station_fuel_type_map[station.station_code] = fuel_type
+
+            wfwx_stations = await get_wfwx_stations_from_station_codes(
+                session, header, list(fire_centre_station_code_ids))
+
+            wfwx_station_ids = [wfwx_station.wfwx_id for wfwx_station in wfwx_stations]
+            raw_dailies_generator = await get_raw_dailies_in_range_generator(
+                session, header, wfwx_station_ids, start_timestamp, end_timestamp)
+            dailies_generator = station_daily_generator(
+                raw_dailies_generator, wfwx_stations, station_fuel_type_map)
+            dailies = []
+            async for station_daily in dailies_generator:
+                dailies.append(station_daily)
+
+            prep_delta = valid_end_date - valid_start_date  # num prep days is inclusive
+
+            results = calculate_hfi_results(request.planning_area_fire_starts,
+                                            dailies, prep_delta.days,
+                                            request.selected_station_code_ids,
+                                            area_station_map,
+                                            valid_start_date)
         response = HFIResultResponse(
-            selected_prep_date=selected_prep_date,
             start_date=start_timestamp,
             end_date=end_timestamp,
             selected_station_code_ids=request.selected_station_code_ids,
             planning_area_station_info=request.planning_area_station_info,
             selected_fire_center_id=request.selected_fire_center_id,
             planning_area_hfi_results=results,
-            planning_area_fire_starts=request.planning_area_fire_starts)
-
-        if request.save is True:
-            save_request(request, token.get('preferred_username', None))
-
+            planning_area_fire_starts=request.planning_area_fire_starts,
+            request_saved=False)
+
+        request_saved = False
```

Comment:
It might be easier to move all the request loading/flagging into it's own class that handles this stuff. Would make it easier to test and remove some complexity from this route

Reply:
Same as above - I've added todo's for it.